src/web-components.js

```diff
@@ -48,6 +48,28 @@ export function isShadowDomSupported() {
 }
 
 /**
+ * Returns `true` if Shadow CSS encapsulation is supported.
+ * @param {!Element} shadowRoot - a shadowRoot element created by createShadowRoot in shadowEmbed.js
+ * @return {boolean}
+ */
+export function isShadowCssSupported(shadowRoot) {
+  if (!isShadowDomSupported()) {
+    return false;
+  }
+
+  // DO NOT SUBMIT
+  //TODO: Find a better way to test CSS encapsulation
+  if (shadowRoot.toString() !== '[object ShadowRoot]') {
```

Comment:
Another idea that would not required `shadowRoot` instance and might be better is something like this:

```
isShadowCssSupported =
    isNative(Element.prototype.attachShadow) ||
    isNative(Element.prototype.createShadowRoot);
```

where, 
```
function isNative(f) {
  return f.toString().indexOf('[native code]') != -1;
}
```

Ugly, but it looks like everything we try here is ugly.

The reasoning: we can conclude that whatever non-native SD implementation is - it will suffer from the same CSS isolation issues.